zio-http/src/main/scala/zhttp/service/ChannelFactory.scala

```diff
@@ -14,19 +13,19 @@ object ChannelFactory {
   def embedded: ZLayer[Any, Nothing, ChannelFactory] = Live.embedded.toLayer
 
   object Live {
-    def nio: UIO[jChannel.ChannelFactory[jChannel.Channel]] =
-      UIO(() => new jSocket.nio.NioSocketChannel())
+    def nio: UIO[JChannel] =
+      UIO(new JNioSocketChannel())
 
-    def epoll: UIO[jChannel.ChannelFactory[jChannel.Channel]] =
-      UIO(() => new jChannel.epoll.EpollSocketChannel())
+    def epoll: UIO[JChannel] =
+      UIO(new JEpollSocketChannel())
 
-    def kQueue: UIO[jChannel.ChannelFactory[jChannel.Channel]] =
-      UIO(() => new jChannel.kqueue.KQueueSocketChannel())
+    def kQueue: UIO[JChannel] =
+      UIO(new JKQueueSocketChannel())
 
-    def embedded: UIO[jChannel.ChannelFactory[jChannel.Channel]] =
-      UIO(() => new jEmbedded.EmbeddedChannel(false, false))
+    def embedded: UIO[JChannel] =
+      UIO(new JEmbeddedChannel(false, false))
 
-    def auto: UIO[jChannel.ChannelFactory[jChannel.Channel]] =
+    def auto: UIO[JChannel] =
```

Comment:
This was actually single abstract method syntax being used to create a factory. There return type should remain as `ChannelFactory`

Reply:
Yeah Scala3 was not happy with that, it was giving me things like
```
[error]  found   : io.netty.channel.embedded.EmbeddedChannel
[error]  required: io.netty.channel.ChannelFactory[io.netty.channel.Channel]
```
I'll do some digging and see if I can figure out why it doesn't resolve that the same way in scala3

Reply:
ok I think I have a work around and will push an update shortly